atlassian_jwt_auth/contrib/django/decorators.py

```diff
@@ -0,0 +1,60 @@
+from functools import wraps
+
+import atlassian_jwt_auth
+from django.conf import settings
+from django.http.response import HttpResponse
+from requests.exceptions import (HTTPError, ConnectionError)
+from jwt.exceptions import (InvalidIssuerError, InvalidTokenError)
+
+from atlassian_jwt_auth.contrib.django.utils import parse_jwt, verify_issuers
+
+
+def requires_asap(issuers):
+    """Decorator for Django endpoints to require ASAP
+
+    :param list issuers: (required) The 'iss' claims that this endpoint is from
+    """
+    def requires_asap_decorator(func):
+        @wraps(func)
+        def requires_asap_wrapper(request, *args, **kwargs):
+            verifier = _get_verifier()
+            auth = request.META.get('AUTHORIZATION', '').split(' ')
+            print(request.META, auth)
+            if not auth or len(auth) != 2:
+                return HttpResponse('Unauthorized', status=401)
+
+            try:
+                asap_claims = parse_jwt(verifier, auth[1])
+                verify_issuers(asap_claims, issuers)
+                request.asap_claims = asap_claims
+                return func(request, *args, **kwargs)
+            except HTTPError:
```

Comment:
I suggest setting an error message based upon the exception type and then having a check to see if the error message has been set and return a HttpResponse like so ->

```
if error:
       return HttpResponse(error, status=401)
```